ipykernel/eventloops.py

```diff
@@ -36,6 +36,12 @@ def process_stream_events():
         if stream.flush(limit=1):
             notifier.setEnabled(False)
             kernel.app.quit()
+        else:
+            # Even if there's nothing to flush, we need to disable the
+            # notifier in order to connect a new one in the next
+            # execution. This applies to the control channel.
+            notifier.setEnabled(False)
+            kernel.app.quit()
```

Comment:
Doesn't putting this in an `else` make the entire `if/else` block unnecessary (both True/False cases run the exact same code)?

Reply:
I though about that too, but the `if` also flushes the shell channel. So perhaps I should rewrite this to

```python
stream.flush(limit=1)
notifier.setEnabled(False)
kernel.app.quit()
```

?